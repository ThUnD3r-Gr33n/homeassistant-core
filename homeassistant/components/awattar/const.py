"""Constants for the aWATTar integration."""

from __future__ import annotations

import logging
from typing import Final

DOMAIN: Final = "awattar"
LOGGER = logging.getLogger(__package__)
