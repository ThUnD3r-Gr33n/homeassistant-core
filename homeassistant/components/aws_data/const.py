"""Constants for the AWS Data integration."""

import logging

DOMAIN = "aws_data"
DOMAIN_DATA = "aws_client"
CONST_FILTER = "filter"
CONST_SERVICE_NAME = "name"
CONST_ACCOUNT = "account"
CONST_SERVICE_ID = "id"
CONST_SERVICE_REASON = "reason"
CONST_INTERVAL = "interval"
CONST_SECONDS = "seconds"
CONST_AWS_KEY = "aws_key"
CONST_AWS_SECRET = "aws_secret"
CONST_AWS_REGION = "aws_region"
CONST_AWS_SERVICES = "aws_services"
CONST_CE_SELECT = "aws_ce_select"
CONST_SCAN_REGIONS = "aws_scan_regions"
CONST_GENERIC_ID = "aws_id_generic"
CONST_ACCOUNT_ID = "aws_account_id"
CONST_ERRORS = "aws_errors"
SERVICE_EC2 = "ec2"
SERVICE_S3 = "s3"
SERVICE_CE = "ce"
EC2_DEF_INTERVAL = 300
S3_DEF_INTERVAL = 86400
CE_DEF_INTERVAL = 86400

GENERAL_REGION = "us-east-1"
DEFAULT_REGION = GENERAL_REGION

EC2_METRIC_STATISTIC = [
    {"metric": "CPUUtilization", "unit": "Percent"},
    {"metric": "NetworkOut", "unit": "Bytes"},
    {"metric": "EBSWriteBytes", "unit": "Bytes"},
    {"metric": "EBSReadBytes", "unit": "Bytes"},
]
S3_METRIC_STATISTIC = [
    {"metric": "s3_objects", "unit": "count"},
    {"metric": "s3_size", "unit": "Bytes"},
]
CE_METRIC_STATISTIC = [
    {"metric": "monthly_cost", "unit": "sum"},
]
SUPPORTED_SERVICES = [SERVICE_EC2, SERVICE_S3, SERVICE_CE]
SUPPORTED_METRICS = {
    SERVICE_EC2: EC2_METRIC_STATISTIC,
    SERVICE_S3: S3_METRIC_STATISTIC,
    SERVICE_CE: CE_METRIC_STATISTIC,
}

API_DATA = "api"
USER_INPUT = "USER_DATA"
U_ID = "INPUT_ID"
U_SECRET = "INPUT_SECRET"
U_REGIONS = "INPUT_REGIONS"
U_SERVICES = "INPUTE_SERVICES"


_LOGGER = logging.getLogger(__name__)
