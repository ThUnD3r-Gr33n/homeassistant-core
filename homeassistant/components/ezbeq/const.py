"""Constants for the ezbeq Profile Loader integration."""

DOMAIN = "ezbeq"
DEFAULT_PORT = 8080
DEFAULT_NAME = "EzBEQ"
CONF_SOURCE_TYPE = "source_type"
CONF_SOURCE_MEDIA_PLAYER = "source_media_player"
CONF_TMDB_SENSOR = "tmdb_sensor"
CONF_YEAR_SENSOR = "year_sensor"
CONF_CODEC_SENSOR = "codec_sensor"
CONF_CODEC_EXTENDED_SENSOR = "codec_extended_sensor"
CONF_EDITION_SENSOR = "edition_sensor"
CONF_TITLE_SENSOR = "title_sensor"
CONF_PREFERRED_AUTHOR = "preferred_author"
CONF_JELLYFIN_CODEC_SENSOR = "jellyfin_codec_sensor"
CONF_JELLYFIN_DISPLAY_TITLE_SENSOR = "jellyfin_display_title_sensor"
CONF_JELLYFIN_PROFILE_SENSOR = "jellyfin_profile_sensor"
CONF_JELLYFIN_LAYOUT_SENSOR = "jellyfin_layout_sensor"
