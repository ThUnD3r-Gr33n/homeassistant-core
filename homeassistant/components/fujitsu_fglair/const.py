"""Constants for the Fujitsu HVAC (based on Ayla IOT) integration."""

from datetime import timedelta

API_TIMEOUT = 10
API_REFRESH = timedelta(minutes=5)

DOMAIN = "fujitsu_fglair"

CONF_EUROPE = "is_europe"
