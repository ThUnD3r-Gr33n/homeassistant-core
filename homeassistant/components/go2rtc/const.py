"""Go2rtc constants."""

DOMAIN = "go2rtc"
