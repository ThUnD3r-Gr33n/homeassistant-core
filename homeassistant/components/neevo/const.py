"""Constants for the Nee-Vo Tank Monitoring integration."""

from typing import Final

DOMAIN: Final = "neevo"
TANKS: Final = "tanks"
COORDINATOR: Final = "coordinator"
