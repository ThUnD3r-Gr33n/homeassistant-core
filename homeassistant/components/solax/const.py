"""Constants for the solax integration."""

DOMAIN = "solax"

MANUFACTURER = "SolaX Power"

CONF_SOLAX_INVERTER = "solax.inverters"

SOLAX_ENTRY_POINT_GROUP = "solax.inverter"
