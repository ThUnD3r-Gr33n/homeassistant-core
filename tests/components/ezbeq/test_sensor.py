"""Tests for the ezbeq sensors."""

from unittest.mock import AsyncMock, patch

import pytest
from syrupy import SnapshotAssertion

from homeassistant.const import STATE_UNAVAILABLE, Platform
from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er

from .conftest import setup_integration

from tests.common import MockConfigEntry, snapshot_platform

pytestmark = pytest.mark.asyncio


async def test_sensor_setup_and_update(
    hass: HomeAssistant,
    mock_ezbeq_client: AsyncMock,
    mock_config_entry: MockConfigEntry,
    entity_registry: er.EntityRegistry,
    snapshot: SnapshotAssertion,
) -> None:
    """Test setup of sensor and data update."""
    with patch("homeassistant.components.madvr.PLATFORMS", [Platform.SENSOR]):
        await setup_integration(hass, mock_config_entry)

    entity_id = "sensor.ezbeq_current_profile"
    entry = entity_registry.async_get(entity_id)
    assert entry
    assert entry.unique_id == f"{mock_ezbeq_client.server_url}_current_profile"

    state = hass.states.get(entity_id)
    assert state
    assert state.state == "Test Profile"

    await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)

    # Simulate a data update
    mock_ezbeq_client.current_profile = "New Test Profile"
    await hass.config_entries.async_reload(mock_config_entry.entry_id)
    await hass.async_block_till_done()

    state = hass.states.get(entity_id)
    assert state
    assert state.state == "New Test Profile"

    # Simulate a data update with unavailable data
    mock_ezbeq_client.current_profile = None
    await hass.config_entries.async_reload(mock_config_entry.entry_id)
    await hass.async_block_till_done()

    state = hass.states.get(entity_id)
    assert state
    assert state.state == STATE_UNAVAILABLE
