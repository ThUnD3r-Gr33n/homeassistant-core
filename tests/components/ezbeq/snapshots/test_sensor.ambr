# serializer version: 1
# name: test_sensor_setup_and_update[sensor.ezbeq_current_profile-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'sensor',
    'entity_category': None,
    'entity_id': 'sensor.ezbeq_current_profile',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Current Profile',
    'platform': 'ezbeq',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': None,
    'unique_id': 'http://192.168.1.100:8080_current_profile',
    'unit_of_measurement': None,
  })
# ---
# name: test_sensor_setup_and_update[sensor.ezbeq_current_profile-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'EzBEQ Current Profile',
    }),
    'context': <ANY>,
    'entity_id': 'sensor.ezbeq_current_profile',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'Test Profile',
  })
# ---
